core/domain/interaction_jobs_one_off.py

```diff
@@ -206,3 +206,189 @@ def map(item):
+            in choices.
+
+            Args:
+                value_type: str. The type of the value.
+                value: *. The value to migrate.
+                choices: list(dict). The list of subtitled html dicts to find
+                    content ids from.
+
+            Returns:
+                *. The migrated rule input.
+            """
+            invalid_values = []
+
+            if value_type == 'DragAndDropHtmlString':
+                if value not in choices:
+                    invalid_values.append(value)
+
+            if value_type == 'SetOfHtmlString':
+                for html in value:
+                    invalid_values.extend(
+                        get_invalid_values(
+                            'DragAndDropHtmlString', html, choices))
+
+            if value_type == 'ListOfSetsOfHtmlStrings':
+                for html_set in value:
+                    invalid_values.extend(
+                        get_invalid_values(
+                            'SetOfHtmlString', html_set, choices))
+
+            return invalid_values
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for state_name, state in exploration.states.items():
+            if state.interaction.id not in [
+                    'DragAndDropSortInput', 'ItemSelectionInput']:
+                continue
+
+            choices = [
+                choice.html
+                for choice in state.interaction.customization_args[
+                    'choices'].value
+            ]
+            solution = state.interaction.solution
+
+            if solution is not None:
+                if state.interaction.id == 'ItemSelectionInput':
+                    invalid_values = get_invalid_values(
+                        'SetOfHtmlString',
+                        solution.correct_answer,
+                        choices)
+                    if invalid_values:
+                        yield (
+                            exploration.id,
+                            (
+                                '<ItemSelectionInput Answer> '
+                                'State: %s, Invalid Values: %s' % (
+                                    state_name, invalid_values)
+                            ).encode('utf-8')
+                        )
+
+                if state.interaction.id == 'DragAndDropSortInput':
+                    invalid_values = get_invalid_values(
+                        'ListOfSetsOfHtmlStrings',
+                        solution.correct_answer,
+                        choices)
+                    if invalid_values:
+                        yield (
+                            exploration.id,
+                            (
+                                '<DragAndDropSortInput Answer> '
+                                'State: %s, Invalid Values: %s' % (
+                                    state_name, invalid_values)
+                            ).encode('utf-8')
+                        )
```

Comment:
Can be probably generalized to (the migh be mistakes in the code I did not test it): 
```suggestion
                interaction_to_value_type = {
                    'ItemSelectionInput': 'SetOfHtmlString',
                    'DragAndDropSortInput': 'ListOfSetsOfHtmlStrings'
                }
                invalid_values = get_invalid_values(
                    interaction_to_value_type[state.interaction.id],
                    solution.correct_answer,
                    choices)
                if invalid_values:
                    yield (
                        exploration.id,
                        (
                            '<%s Answer> '
                            'State: %s, Invalid Values: %s' % (
                                state.interaction.id, state_name, invalid_values)
                        ).encode('utf-8')
                    )
```